### DISCUSSION TITLE
Include event.object in spy reports when using store method as a bound function

### DISCUSSION THREAD

**Discussion by kubk** [mobxjs/mobx#3140] [General] [answered]
It is more than feature request than a bug that's why I want to discuss it first. Suppose we have code like this:

```js
class AppState {
  value = 0;

  constructor() {
    makeAutoObservable(this, {}, { autoBind: true });
  }

  increase() {
    this.value++;
  }
}

const state = new AppState();

spy((event) => {
  if (event.type === 'action') {
    console.log('spy', event);
  }
});

const TimerView = observer(() => {
  return <div>
    <button onClick={() => state.increase()}>Value: {state.value}</button>
    <button onClick={state.increase}>Value: {state.value}</button>
  </div>
});
```

I expect `spy` to behave the same when calling `increase` as a class method and as a bound function. But if I click the first button and then second button I get the following result:

```
type: "action"
name: "increase"
object: AppState // <- key 'object' is not empty, it's correct 
arguments: Array(0)
spyReportStart: true
```

```
type: "action"
name: "increase"
object: undefined // <- key 'object' is empty. Which store caused the action? It's unclear
arguments: Array(1)
spyReportStart: true
```

I am developing a logging library for Mobx. I'd be nice to have this property not empty to display more information about action for user. Is it possible to add it here?

Reproduction: https://codesandbox.io/s/interesting-cookies-3p31q?file=/index.js

**Comment by urugator** [comment, marked as answer]
Done https://github.com/mobxjs/mobx/pull/3141 :)

**Comment by kubk** [comment]
Thank you @urugator  ! :+1:
